src/Swarm/Game/Step.hs

```diff
@@ -374,10 +374,19 @@ stepCESK cesk = case cesk of
   -- Now some straightforward cases.  These all immediately turn
   -- into values.
   In TUnit _ s k -> return $ Out VUnit s k
-  In (TDir d) _ s k -> return $ Out (VDir d) s k
   In (TInt n) _ s k -> return $ Out (VInt n) s k
   In (TString str) _ s k -> return $ Out (VString str) s k
   In (TBool b) _ s k -> return $ Out (VBool b) s k
+  -- Direction is simple too unless it is absolute (e.g. north)
+  -- and needs orient capability (provided by a compass).
+  -- This may be redundant just because of base escaping
+  -- capability checking (see #231).
+  In (TDir d) _ s k -> do
+    orient <- hasCapability COrient
+    if isCardinal d && not orient
+      then return $ Up (Incapable (S.singleton COrient) (TDir d)) s []
+      else return $ Out (VDir d) s k
```

Comment:
I am not sure about this.  Should we really need the `orient` capability just to *evaluate* a cardinal direction constant?

Reply:
Well, `base` would escape the check if I did not do it, but it still seems wrong:
```Haskell
> turn north -- compass is needed (but the error does not mention it)
> let d = north in build {turn d} -- once base has compass, nobody else needs it
```
I probably just overthought it and checking it when turning and drilling would work just fine.

- related #201 for the case when a robot is sent in some direction and it turns out it is cardinal at runtime (I should write a test for it...)
- related error message improvement #342

Reply:
Reverted and moved the check to commands using the direction.